proposals/large-crypto-blobs.md

```diff
@@ -0,0 +1,127 @@
+
+In `js-libp2p` and `js-ipfs`, we have been working towards reducing the bundle size of the library to the minimum possible, so that the users of our stack don't need to ship gigantic bundle size applications just for using parts of our stack. However, we also created technical debt when `libp2p-noise` was added to the stack, as a new crypto library was added.
+
+One of the outstanding issues to have our bundle size in a considerably good shape is the crypto libraries that we need to carry. The JS ecosystem around cryptography is far from perfect, which has made us rely on several different libraries to achieve different cryptographic primitives.
+
+![image info](./js-libp2p-bundle.png)
+
+The current bundle size for libp2p gzipped is 222.63KB. From the image above we can see that `node-forge`, `elliptic`, `libp2p-crypto`, `hash.js` and `blakejs` consist of around half of our shipped bundle.
+
+Taking into account that libp2p is a composable and modular stack, most of the times `js-libp2p` bundle contains heavy crypto code that is not needed. In this context, `js-libp2p` crypto should be modular and pluggable, so that the users can bring just what they need instead of the all thing.
+
+This has been a pain for some users, including team spark did not rely on libp2p for NFT project because it was too heavy. This can be a threat for the wide usage of libp2p in the web3 ecosystem.
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Web3 users want to create libp2p backed applications with only the code they use bundled
+- Web3 users want to provide smooth and fast experiences to their users
+- Web3 users do not want to have their apps taking X seconds to load
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+- Users can configure via libp2p configuration what cryptographic modules they are interested in using
+- Libp2p provides custom bundles for the most common use cases that users can rely on (out of the box usage)
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
```

Comment:
I would suggest  to put crypto interface definition that can be evaluated fairly cheaply as a first step. I think js-multiformats does a good job doing it for [multibase](https://github.com/multiformats/js-multiformats/blob/master/src/bases/interface.ts), [multihash](https://github.com/multiformats/js-multiformats/blob/master/src/hashes/interface.ts), [multicodec](https://github.com/multiformats/js-multiformats/blob/master/src/codecs/interface.ts) and I would love similar thing for crypto.